### ISSUE TITLE
sbt should issue a warning about ignoring javaOptions when fork is set to false in build.sbt

### ISSUE THREAD

**Issue by stammana** [sbt/sbt#2041]
The following is not being honored when fork is set to false.

javaOptions in Test += "-Dconfig.file=conf/application_test.conf"

It works fine when the fork parameter is removed from build.sbt

sbt version 0.13.5.

**Comment by pdalpra** [comment]
From SBT itself: 

```
> help javaOptions
Options passed to a new JVM when forking.
```

`javaOptions` is only used when forking new JVMs.

**Comment by stammana** [comment]
the fork is set to false not to true. what is the default value for fork in sbt?

**Comment by pdalpra** [comment]
Again, from SBT itself : 

```
> show test:fork
[info] false
```

SBT does not fork new JVMs by default.

**Comment by stammana** [comment]
> show test:fork
> [info] true

Getting it as true by default on version 0.13.5

**Comment by pdalpra** [comment]
It must have been overriden someplace else in your build, possibly from a plugin, as `false` is indeed the default value : https://github.com/sbt/sbt/blob/0.13.5/main/src/main/scala/sbt/Defaults.scala#L152

**Comment by stammana** [comment]
seems like you are correct, but why is it not honoring my javaOptions when fork is set to false.

**Comment by pdalpra** [comment]
See my previous comment. `javaOptions` is only used when forking new JVMs. If `fork in Test` is set to `false`, `javaOptions in Test` is simply not used, since you're not forking new JVMs.

**Comment by stammana** [comment]
It would have been great if it takes java options when fork is set to false. I don't need to supply it from the command line. I can simply say "sbt clean test". 

Can this be an enhancement?

**Comment by jsuereth** [comment]
@stammana Not really.   By the time sbt has read your build file, the JVM is already instantiated and we can't send javaOptions to it.  The only way to send java options is to fork a new JVM where you can pass them in, you cannot set them after it's been started.

So, you can either set javaOptions to true, or you can use one of the startup script tricks of creating a .sbtopts file and putting your java options in there.

At a maximum, we'll just issue a warning when people do this.

**Comment by dwijnand** [comment]
Closed with #2103
